File: makeMyTour/my-app/src/services/refund.ts
import api from "@/lib/axios";

export interface RefundData {
  id: string;
  userEmail: string;
  bookingId: string;
  bookingType: string;
  originalAmount: number;
  refundAmount: number;
  refundPercent: number;
  reason: string;
  status: string;
  requestedAt: string;
  processedAt: string | null;
  expectedCompletionDate: string;
}

export const getCancellationReasons = () =>
  api.get<string[]>("/api/refunds/reasons");

export const cancelBooking = (email: string, bookingId: string, reason: string) =>
  api.post<RefundData>("/api/refunds/cancel", { email, bookingId, reason });

export const getRefunds = (email: string) =>
  api.get<RefundData[]>(`/api/refunds/${email}`);

File: makeMyTrip/src/main/java/com/MakeMyTrip/makeMyTrip/models/Refund.java
package com.MakeMyTrip.makeMyTrip.models;

import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;

@Document(collection = "refunds")
public class Refund {
    @Id
    private String id;
    private String userEmail;
    private String bookingId;
    private String bookingType;
    private double originalAmount;
    private double refundAmount;
    private double refundPercent;
    private String reason;
    private String status;
    private String requestedAt;
    private String processedAt;
    private String expectedCompletionDate;

    public String getId() { return id; }
    public void setId(String id) { this.id = id; }
    public String getUserEmail() { return userEmail; }
    public void setUserEmail(String userEmail) { this.userEmail = userEmail; }
    public String getBookingId() { return bookingId; }
    public void setBookingId(String bookingId) { this.bookingId = bookingId; }
    public String getBookingType() { return bookingType; }
    public void setBookingType(String bookingType) { this.bookingType = bookingType; }
    public double getOriginalAmount() { return originalAmount; }
    public void setOriginalAmount(double originalAmount) { this.originalAmount = originalAmount; }
    public double getRefundAmount() { return refundAmount; }
    public void setRefundAmount(double refundAmount) { this.refundAmount = refundAmount; }
    public double getRefundPercent() { return refundPercent; }
    public void setRefundPercent(double refundPercent) { this.refundPercent = refundPercent; }
    public String getReason() { return reason; }
    public void setReason(String reason) { this.reason = reason; }
    public String getStatus() { return status; }
    public void setStatus(String status) { this.status = status; }
    public String getRequestedAt() { return requestedAt; }
    public void setRequestedAt(String requestedAt) { this.requestedAt = requestedAt; }
    public String getProcessedAt() { return processedAt; }
    public void setProcessedAt(String processedAt) { this.processedAt = processedAt; }
    public String getExpectedCompletionDate() { return expectedCompletionDate; }
    public void setExpectedCompletionDate(String expectedCompletionDate) { this.expectedCompletionDate = expectedCompletionDate; }
}
